apps/web/composables/useCategoryTemplate/useCategoryTemplate.ts

```diff
@@ -6,39 +6,80 @@ import type {
   SaveBlocks,
 } from '~/composables/useCategoryTemplate/types';
 import type { Block } from '@plentymarkets/shop-api';
+import type { FooterSettings } from '~/components/blocks/Footer/types';
 
 import homepageTemplateDataDe from './homepageTemplateDataDe.json';
 import homepageTemplateDataEn from './homepageTemplateDataEn.json';
 
 const useLocaleSpecificHomepageTemplate = (locale: string) =>
   locale === 'de' ? (homepageTemplateDataDe as Block[]) : (homepageTemplateDataEn as Block[]);
 
-export const useCategoryTemplate: UseCategoryTemplateReturn = () => {
-  const state = useState<UseCategoryTemplateState>('useCategoryTemplate', () => ({
+export const useCategoryTemplate: UseCategoryTemplateReturn = (blocks?: string) => {
+  const state = useState<UseCategoryTemplateState>(`useCategoryTemplate-${blocks}`, () => ({
     data: [],
     cleanData: [],
     categoryTemplateData: null,
     loading: false,
   }));
 
-  const getBlocks: GetBlocks = async (identifier, type) => {
+  const { $i18n } = useNuxtApp();
+
+  const getBlocksServer: GetBlocks = async (identifier, type, blocks?) => {
     state.value.loading = true;
-    const { data, error } = await useAsyncData(`${type}-${identifier}`, () =>
-      useSdk().plentysystems.getBlocks({ identifier, type }),
+
+    const { data, error } = await useAsyncData(`${type}-${identifier}-${blocks}`, () =>
+      useSdk().plentysystems.getBlocks({ identifier, type, ...(blocks ? { blocks } : {}) }),
```

Comment:
```suggestion
      useSdk().plentysystems.getBlocks({ identifier, type, blocks }),
```
**question**: Why do we have to destructure `blocks` here? `blocks` should always be of type string, so there shouldn't be anything to destructure.